### ISSUE TITLE
Option to ignore contig coverage

### ISSUE THREAD

**Issue by jmtsuji** [RasmussenLab/vamb#18]
Hello @jakobnissen 

Thanks for developing this interesting tool. I'm currently working with an unusual dataset where contig coverage information is not helpful for generating genome bins. (We manipulated the DNA samples in the wet lab prior to sequencing as part of testing a new method; this threw off relative abundances of DNA fragments.) As such, I'm trying to find a nicely packaged algorithm that can generate genome bins solely using tetranucleotide frequencies.

Vamb seems like it should be able to handle working with tetranucleotide frequencies alone, given that it treats input data flexibly (i.e., can handle heteromorphic data). Most other binning tools that I've looked into seem to strictly require both tetranucleotide frequency and contig coverage (I assume this is hard-coded into the algorithm).

Is it possible to use vamb without contig coverage information? If so, would it be possible to make a command line flag for this (e.g., `--no_coverage`)? I understand that this is probably not recommended for standard metagenomic data, but it could be quite helpful for my case (and perhaps demonstrate the flexibility of the tool).

Thanks.

**Comment by jakobnissen** [comment]
Dear @jmtsuji 

We are currently running more tests on Vamb, including its performance on TNF only and depths only. While we're not done yet, it appears to be the case that most information is found in the co-abundance, and relatively less information in TNF. So, using only TNF information leads to poor binning.

In your case, I would recommend binning your sequences by aligning them to a database of some sort and then simply pick out the sequences that align to the same reference. However, if you happen to know that your sequences are mostly not in any database, for example if you are sequencing a complex environment like soil or marine sediment, that might not be possible.

You CAN run Vamb without the depths information, but you would need to do it from within a Python interpreter. See the `docs/tutorial.ipynb`, where a more complete description of Vamb can be found also. In your particular case, I would manually create a depth matrix with uniform values, for example an (n_sequences x 4) matrix filled with `1.0`s. Then, proceed as normally in Vamb. The network will quickly learn to perfectly reconstruct the "abundances", after which it will focus on encoding the TNF. You might also lower the number of latent neurons to e.g. 32 or something.

To recap: I don't recommend using only TNF, but of course I don't know your exact situation. I will not add a `--no-coverage` flag, because I wouldn't recommend using it in the first place. Furthermore, I believe it's a more sustainable solution to keep the Python "internals" of Vamb documented and transparent, such that users can manipulate the program in ways which I cannot predict. This is because the amount of possible variations on how Vamb could be run could quickly swamp the development in feature creep.

**Comment by jmtsuji** [comment]
Hello @jakobnissen 

This is very helpful -- thanks for the detailed response.

It sounds like genome binning based on TNF information alone has very poor performance. This is good for me to know. Also, thanks for the tip of how to perform binning from within the Python interpreter. I agree with your design philosophy to keep the vamb internals transparent but have recommended workflows possible through the command line.

I might try the Python interpreter method, but given your advice, it might not be worth it to try purely TNF-guided binning methods at all.

Nice work on this tool -- I look forward to seeing its continued development for use with the typical metagenomes I work with.

Jackson
